build_runner/lib/src/asset/reader.dart

```diff
@@ -2,18 +2,34 @@ import 'dart:async';
 import 'dart:convert';
 
 import 'package:build/build.dart';
+import 'package:crypto/crypto.dart';
 import 'package:glob/glob.dart';
 import '../asset_graph/graph.dart';
 import '../asset_graph/node.dart';
 
 typedef Future RunPhaseForInput(int phaseNumber, AssetId primaryInput);
 
+abstract class DigestAssetReader implements AssetReader {
+  /// Asynchronously compute the digest for [id].
```

Comment:
should we document expected behavior of this call if the Asset does not exist?

Reply:
done